docs/okeps/okep-5193-user-defined-networks.md

```diff
@@ -0,0 +1,1468 @@
+
+    ```
+   sh-5.2# ip rule
+   
+   0:	from all lookup local
+   30:	from all fwmark 0x1745ec lookup 7
+   5999:	from all fwmark 0x3f0 lookup main
+   6000:	from 10.244.2.3 lookup 1111
+   32766:	from all lookup main
+   32767:	from all lookup default
+   ```
+
+3. Iptables rules save the packet mark in conntrack. This is only applicable to packets that were marked with 1008 and
+   are bypassing egress IP:
+
+    ```
+    sh-5.2# iptables -t mangle -L  PREROUTING
+
+    Chain PREROUTING (policy ACCEPT)
+    target     prot opt source               destination
+    CONNMARK   all  --  anywhere             anywhere             mark match 0x3f0 CONNMARK save
+    CONNMARK   all  --  anywhere             anywhere             mark match 0x0 CONNMARK restore
+    ```
+
+4. VRF 1111 has a route in it to steer the packet to the right egress interface:
+
+    ```
+    sh-5.2# ip route show table 1111
+    default dev eth1
+    ```
+
+5. IPTables rules in NAT table SNAT the packet:
+
+    ```
+    -A OVN-KUBE-EGRESS-IP-MULTI-NIC -s 10.244.2.3/32 -o eth1 -j SNAT --to-source 10.10.10.100
+    ```
+
+6. For reply bypass traffic, the 0x3f0 mark is restored, and ip rules 5999 send it back into default VRF for routing
+   back into mp0 for non-egress IP packets. This is rule and connmark restoring is required for the packet to pass the
+   reverse path filter (RPF) check. For egress IP reply packets, there is no connmark restored and the packets hit the
+   default routing table to go back into mp0.
+
+This functionality will continue to work, with ip rules steering the packets from the per network VRF to the appropriate
+egress IP VRF. CONNMARK will continue to be used so that return traffic is sent back to the correct VRF. Step 5 in the
+above may need to be tweaked to match on mark in case 2 pods have overlapping IPs, and are both egressing
+the same interface with different Egress IPs. The flow would look something like this:
+
+![Egress IP VRF LGW](../images/egress-ip-vrf-lgw.svg)
+
+##### Egress Firewall
+
+Egress firewall is enforced at the OVN logical switch, and this proposal has no effect on its functionality.
+
+
+##### Egress QoS
+
+Egress QoS is namespace scoped and functions by marking packets at the OVN logical switch, and this proposal has no
+effect on its functionality.
+
+##### Egress Service
+
+Egress service functions similar to Egress IP in local gateway mode, with the exception that all traffic paths go
+through the kernel networking stack. Egress Service also uses IP rules and VRFs in order to match on traffic and forward
+it out the right network (if specified in the CRD). It uses iptables in order to SNAT packets to the load balancer IP.
+Like Egress IP, with user-defined networks there will need to be IP rules with higher precedence to match on packets from
+specific networks and direct them to the right VRF.
+
+##### Multiple External Gateways (MEG)
```

Comment:
Here and everywhere, I think this feature is now called AdminPolicyBasedRoutes or something but I won't hold the enhancement merge for that.